README.md

```diff
@@ -61,8 +61,111 @@ This repository contains the following stand-alone applications, which demonstra
 
 ## Tiny Typir Example
 
-[TODO](/packages/typir/test/api-example.test.ts)
-
+Both the LOX and OX examples have been created with Langium. Here comes a very small Langium independent example for the usage of Typir with a tiny expression language. We show how to use the Typir API for its Tiny Typir type checking. You can also find the example in the repository code, implemented in form of a [test](/packages/typir/test/api-example.test.ts).
+
+Our Tiny Typir language has only a few possible AstElements, namely NumberLiteral, StringLiteral, BinaryExpression, Variable, and AssignmentStatement. They are implemented in a very simple way, see for example our BinaryExpression:
+
+```
+class BinaryExpression extends AstElement {
+    constructor(
+        public left: AstElement,
+        public operator: string,
+        public right: AstElement,
+    ) { super(); }
+}
+```
+
+Feel free to check out the others in the [test code](/packages/typir/test/api-example.test.ts), but a little spoiler: no surprises there.
+
+Let's head into setting up the Typir type system and creating the primitive types for our NumberLiteral and StringLiteral, which is a one line of code job each, as we use the Typir's predefined Primitives factory service:
+
+```
+const typir = createTypirServices();
+
+const numberType = typir.factory.Primitives.create({ primitiveName: 'number', inferenceRules: node => node instanceof NumberLiteral });
+
+const stringType = typir.factory.Primitives.create({ primitiveName: 'string', inferenceRules: node => node instanceof StringLiteral });
+```
+
+Note that the inference rules are included in this. For the operators this is a bit longer, as we have to take care of the left and right operand and the operator of the binary expression, so we
+extract it:
+
+```
+const inferenceRule: InferOperatorWithMultipleOperands<BinaryExpression> = {
+    filter: node => node instanceof BinaryExpression,
+    matching: (node, operatorName) => node.operator === operatorName,
+    operands: node => [node.left, node.right],
+};
+```
+
+We wish to have two operators, the `+` operator, which should be overloaded to accept either two numbers to add or two strings to concatenate. This can be expressed with an array of signatures with different types for the operands and the return type of the operator. Furthermore, there is going to be a `-` operator with only one signature, as we need no overloading. Both refer to the inferenceRule we defined above:
```

Comment:
Both refer to the inferenceRule we defined above. `numberType` and `stringType` are the primitive Typir types we defined above.

Reply:
I added this and also added "Both operands" so that it is clear what it refers to